coderd/jwtutils/jwe.go

```diff
@@ -0,0 +1,127 @@
+package jwtutils
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/cryptokeys"
+)
+
+const (
+	encryptKeyAlgo     = jose.A256GCMKW
+	encryptContentAlgo = jose.A256GCM
+)
+
+// Encrypt encrypts a token and returns it as a string.
+func Encrypt(ctx context.Context, keys cryptokeys.Keycache, claims Claims) (string, error) {
```

Comment:
Could this accept an interface instead of the cryptokeys type? It only needs the Signing method